app/src/components/MemberBox.jsx

```diff
@@ -0,0 +1,235 @@
+  background: none;
+  color: inherit;
+  border: none;
+  padding: 0;
+  font: inherit;
+  cursor: pointer;
+  outline: inherit;
+  color: blue; 
+  text-decoration: underline;
+
+  &:hover{
+    color: darkblue;
+  }
+`;
+
+const Input = styled.input`
+  padding: 5px 8px;
+`;
+
+// INLINE COMPONENTS (ONLY FOR TOGGLING)
+const addMemberForm = {
+  flexDirection: "column",
+  marginTop: "20px",
+  display: "none",
+  height: "120px",
+  border: "1px solid lightgray",
+  justifyContent: "space-between",
+  padding: "10px 20px",
+  background: "whitesmoke"
+}
+
+const formError = {
+  color: 'red',
+  fontSize: '0.9rem',
+  display: 'none',
+}
+
+// TEMPORARY CONSTANT (REPLACED BY AJAX REQUEST)
+const groupMembers = [
+  {name: 'Frodo Baggins', phone: 123444, isChecked: false}, 
+  {name: 'Peregrin  Took', phone: 1234, isChecked: false}, 
+  {name: 'Meriadoc Brandybuck', phone: 12345, isChecked: false}, 
+  {name: 'Famwise Gamgee', phone: 123456, isChecked: false},
+  {name: 'Wrodo Baggins', phone: 1258883, isChecked: false}, 
+  {name: 'Reregrin  Took', phone: 825485, isChecked: false}, 
+  {name: 'Seriadoc Brandybuck', phone: 66672, isChecked: false}, 
+  {name: 'Ramwise Gamgee', phone: 235235, isChecked: false}
+]
+
+class MemberBox extends Component {
+  constructor() {
+    super();
+    this.state = {
+      members: [],
+      name: '',
+      phone: '',
+    }
+  }
+
+  componentDidMount() {
+    // GET request goes here:
+    this.setState({
+      members: groupMembers,
+    })
+  }
+
+  toggleCheckbox = (phone) => {
+    let updatedMembers = [...this.state.members];
+
+    updatedMembers.map(member => {
+      (member.phone === phone &&
+        (member.isChecked = !member.isChecked) 
+      )
+    })
+
+    this.setState({
+      members : updatedMembers,
+    })
+  }
+
+  toggleMemberForm = () => {
+    let memberForm = document.querySelector('.addMemberForm');
+
+    if (memberForm.style.display === "none") {
+      memberForm.style.display = "flex";
+    } else {
+      memberForm.style.display = "none";
+    }
+  }
+
+  handleChange = e => {
+    this.setState({[e.target.name]: e.target.value});
+  }
+
+  handleSubmit = (e) => {
+    e.preventDefault();
+
+    if (this.state.name === "" || this.state.phone === "") {
+      document.querySelector('.formError').style.display = 'block';
+    } else {
+      document.querySelector('.formError').style.display = 'none';
+
+      let newMember = {
+        name: this.state.name, 
+        phone: parseInt(this.state.phone), 
+        isChecked: false
+      };
+
+      let updatedMembers = [...this.state.members];
+
+      updatedMembers.push(newMember);
+  
+      this.setState({
+        members: updatedMembers,
+        name: '',
+        phone: '',
+      })
+    }
+  }
+
+  selectAll = () => {
+    let updatedMembers = [...this.state.members]
+    updatedMembers.map(member => member.isChecked = true)
+
+    let checkboxes = document.getElementsByName('checkbox');
```

Comment:
The main idea with React is that you just manage the data, and you tell React how to render based on the data in the `render` function. You want to be declarative about how things should render. What you usually don't want to do is be "imperative", where you're manipulating the DOM yourself.  See if you can manage this form without any DOM manipulation side effects in your methods

Reply:
Thank you Kamry, your suggestions have really helped me grasp the concept of thinking in terms of the state first. I've refactored all DOM manipulation except: 

```
    let checkboxes = document.getElementsByName('checkbox');
    for(var i=0, n=checkboxes.length;i<n;i++) {
      checkboxes[i].checked = bool;
    }
```

I'm stumped on how to deal with that one but it's minor. Way too tired right now lol.